pkg/lib/oauth/handler/handler_token.go

```diff
@@ -590,7 +590,9 @@ func (h *TokenHandler) IssueTokensForAuthorizationCode(
 			fmt.Sprintf("failed to match dpop jkt on issue tokens: %s", dpopErr.Message),
 			slog.Bool("dpop_logs", true),
 		)
-		return nil, ErrInvalidDPoPKeyBinding
+		if !client.DPoPDisabled {
+			return nil, ErrInvalidDPoPKeyBinding
+		}
```

Comment:
It is better to make `codeGrant.MatchDPoPJKT` takes `client` (`codeGrant.MatchDPoPJKT(client, dpopProof)`) so that MatchDPoPJKT never return error if client.DPoPDisabled is true.

In general, make all places that enforce DPoP takes `client`, instead of conditionally returning error.

- `codeGrant.MatchDPoPJKT`
- `offlineGrant.MatchDeviceSecretDPoPJKT`
- `offlineGrantSession.MatchDPoPJKT`

Reply:
We want to preserve the matching result in logs, only not to return error in response. So I guess doing the check outside is better.
But I think we can have a helper method to avoid code duplication.

Reply:
What about you return struct containing two `error`s?

```golang
type DPoPJKTResult struct {
  ErrorForClient error
  ErrorForLogging error
}

func MatchDPoPJKT(client, proof) DPoPJKTResult
```